odoo/addons/test_lint/tests/_odoo_checker_gettext.py

```diff
@@ -54,6 +59,8 @@ def visit_call(self, node):
                     self.add_message('gettext-variable', node=node)
                 elif len(PLACEHOLDER_REGEXP.findall(str(first_arg.value))) >= 2:
                     self.add_message('gettext-placeholders', node=node)
+                elif re.search(r"%(\(\w+\))?r", first_arg.value):
```

Comment:
compile the regexp, move the compiled object on top and change it to use non-capturing group
```py
r"%(?:\(\w+\))?r"
```